test/README.md

```diff
@@ -1,11 +1,38 @@
 # Guide on testing
 
 - [Test with `TestManager` Module](#test-with-testmanager-module)
+    - [TestManager Overview](#Overview-About-testManager-Module)
     - [Compile](#compile)
-    - [Configure](#configure)
+    - [Configure](#Modify-the-configuration-files-accordingly)
     - [Run](#run)
     - [Verify](#verify)
 
+## Overview About testManager Module
+
+* testManager is a utility module to mimic the cloud and push messages for different kinds of actions that could happen from the cloud. Typical device , node and application lifecycle management functions are expected to be performed in the cloud and pushed to the edge node. These functions commonly encompass configurations related to 
+    
+     - Kubernetes Secrets and Configuration Maps.
+     - Application deployment/sync
+     - Binding devices to edge nodes via memberships.
+     - Syncing of different resources between cloud and edge ( like app status, device status etc)
+     - Node sync, etc..
+     
+Below info can help user how to use the testManager for testing the kubeedge.
+
+* testManager module acts as a http server where it exposes its own api's to access the some of the kubeedge functionality's.
+* testManager will be listening on **http://127.0.0.1:12345** EndPoint.
+* API list exposed by testManager for various operations are:.
+
+    - /pods         : Deploying a pod on to kubeedge node.
+    - /apps         : Deploying a application on to kubeedge node.
+    - /devices      : Bind a Device to kubeedge node.
+    - /secrets      : Configure secrets on kubeedge node.
+    - /configmaps   : Configure configmaps on kubeedge node.
+    
+* Using the above api calls, user can do a resource operations for kubeedge test.
+* testManager provides feasibility to test the kubeedge features just by using **Curl** commands without running the test suites. 
```

Comment:
Curl -> curl. 
rephrase "testManager facilitates validating the capabilities of the edge platform by performing curl operations against a running edge node.